File: Data Structures & Algorithms/longest-consecutive-sequence/submission-6.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        result = 0
        store = set(nums)

        for s in store:
            if (s - 1) not in store:
                streak = 1
                while s + streak in store:
                    streak += 1
                result = max(result, streak)
        return result

File: Data Structures & Algorithms/two-integer-sum/submission-6.rs
impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let nums_size = nums.len();

        // Iterate through every possible pairs in nums
        for i in 0..nums_size {
            for j in (i + 1)..nums_size {
                // Return result if we've found target
                if nums[i] + nums[j] == target {
                    return vec![i as i32, j as i32];
                }
            }
        }

        // Return empty result to satisfy compiler
        // Since the problem says there is guaranteed to be exactly one solution,
        // so we will never return an empty array.
        unreachable!()
    }
}
